### ISSUE TITLE
Bullets (visually) spawn behind the player on highfreq

### ISSUE THREAD

**Issue by derrod** [VeniceUnleashed/VeniceUnleashed#139]
This ia a thing I noticed while playing on Vanilla today.

30 Hz/Vanilla: ![30hz](http://i.imgur.com/votedao.jpg)
120 Hz: ![120hz](http://i.imgur.com/7zzdrSu.jpg)

**Comment by Powback** [comment]
I'm going to assume that the bullet trail is initiated too soon because of the tickrate change. Normally the trail would start after x time, but the tickrate change causes it to be initiated 12 times earlier.

**Comment by OrfeasZ** [comment]
This is only a visual bug and doesn't actually affect gunplay. 

The reason behind this, like @powback correctly suggested, is that due to the high update rate, you get the bullet spawn event much earlier than you normally would. Since bullets are spawned slightly behind the player (possibly to account for instances where collision with another object would prevent the bullet from hitting), you can actually see its trail much sooner than you would in regular freq.

A possible workaround for this would be to "hide" the trail for a few ms after the bullet has been fired, but it's not something we'll be looking into right now.

**Comment by Powback** [comment]
This issue can be fixed with the following mod:

```
	if p_Instance.typeName == "EmitterEntityData" then
		local s_Instance = EmitterEntityData(p_Instance)
		if(s_Instance.startDelay == 0) then
			s_Instance.startDelay = 0.04
		end
	end
```

Bullets spawn 1:1 with vanilla in high120. 

In high60, the bullets spawn a little bit futher ahead, but it's not noticeable. I've also tried values between 0.03 and 0.04, but haven't been able to replicate the vanilla bulletspawn. 0.04 comes the closest, and also fixes the bullets spawning in your face with certain guns.

This can be implemented natively, but I fear that it might override user-made mods unless the native fix is loaded first.

I have not noticed any side-effects since the delay is so small. 

Normal: https://gfycat.com/DrearyDigitalIrishdraughthorse
Bugged: https://gfycat.com/VagueMealyKitty
Fixed: https://gfycat.com/PeskyAlienatedAmericanratsnake
